### PR TITLE
Add support for `.xctestproducts` and Test Plan

### PR DESCRIPTION
**Pull request by DavidAsulin1**
## Description
In this PR, I suggest adding support for the `.xctestproducts` package in XCTest. As I described in issue #583, using `xctestproducts` is more suitable for CI and also supports the Test Plan feature. 

Currently, in order to export the test products, one needs to package the Derived Data and the `.xctestrun` file. After this change, the user can choose to upload the `xctestproducts` package, and we will detect it internally.

One additional change on the UI side is regarding the Test Plan name. In case the test type is XCTest, the user can enter the test plan name, which we will use if the test package is of type `xctestproducts`.

If we decide to proceed with this change, the [Wiki page](https://github.com/microsoft/HydraLab/wiki/Create-test-build-and-run-XCTest) will need to be updated.

## Pull Request Checklist

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Code compiles correctly with all tests are passed.
- [x] I've read the [contributing guide](https://github.com/microsoft/HydraLab/blob/main/CONTRIBUTING.md#making-changes-to-the-code) and followed the recommended practices.
- [ ] [Wikis](https://github.com/microsoft/HydraLab/wiki) or [README](https://github.com/microsoft/HydraLab/blob/main/README.md) have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?
*If this introduces a breaking change for Hydra Lab users, please describe the impact and migration path.*

- [ ] Yes
- [x] No

## How you tested it
I test the change locally by uploading `xctestproducts` package and by choosing different test plans to execute

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Technical design
- [ ] Build related changes
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Code style update (formatting, renaming) or Documentation content changes
- [ ] Other (please describe):

### CODE REVIEW

**Comment by hydraxman** [comment]
@DavidAsulin1 Again, thank you for the contribution, on behalf of the team.

Would you mind adding a UT that can cover the changed branches in the code?

File: common/src/main/java/com/microsoft/hydralab/common/entity/common/TestTask.java [hunk 1]
```diff
@@ -164,6 +165,9 @@ public static TestTask convertToTestTask(TestTaskSpec testTaskSpec) {
         if (StringUtils.isNotBlank(testTaskSpec.runningType)) {
             testTask.setRunningType(testTaskSpec.runningType);
         }
+        if (StringUtils.isNotBlank(testTaskSpec.testPlan)) {
```
Comment (zhou9584) [inline common/src/main/java/com/microsoft/hydralab/common/entity/common/TestTask.java:168]: You need to do the same operation in convertToTestTaskSpec method too. Hydra Lab will retry to trigger test task by convert TestTask to TestTaskSpec if meet exception when running.

**Comment by DavidAsulin1** [comment]
> @DavidAsulin1 Again, thank you for the contribution, on behalf of the team.
> 
> Would you mind adding a UT that can cover the changed branches in the code?

@hydraxman The changes I made are better suited for Integration Testing rather than Unit Testing. I updated the private `runXctest` implementation to handle a different package behavior.

**Review by zhou9584 — APPROVED** [review]

**Review by olivershen-wow — APPROVED** [review]

**Pull request merged**